### PR TITLE
Supports service discovery with Etcd

### LINKED ISSUE #812: Supports service discovery with Etcd
i will do this.

### PR DESCRIPTION
**Pull request by xingfudeshi**
### Ⅰ. Describe what this PR did
Supports service discovery with Etcd

### Ⅱ. Does this pull request fix one issue?

fixes #812 

### Ⅲ. Why don't you add test cases (unit test/integration test)? 
i have done integration test.

### CODE REVIEW

File: discovery/seata-discovery-etcd3/src/main/java/io/seata/discovery/registery/etcd/EtcdRegistryServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,403 @@
...
+        if (null != subscribeSet) {
+            Set<Watch.Listener> newSubscribeSet = new HashSet<>();
+            for (Watch.Listener eventListener : subscribeSet) {
+                if (!eventListener.equals(listener)) {
+                    newSubscribeSet.add(eventListener);
+                }
+            }
+            listenerMap.put(cluster, newSubscribeSet);
+        }
+        watcherMap.remove(cluster).stop();
```
Comment (slievrly) [inline discovery/seata-discovery-etcd3/src/main/java/io/seata/discovery/registery/etcd/EtcdRegistryServiceImpl.java]: can not stop EtcdLifeKeeper#process？

Comment (xingfudeshi) [reply discovery/seata-discovery-etcd3/src/main/java/io/seata/discovery/registery/etcd/EtcdRegistryServiceImpl.java]: > can not stop EtcdLifeKeeper#process？

EtcdLifeKeeper is used to refresh the lease which binding to registry key.like the name,it's for keeping the instance in healthy.I think when calling unsubscribe method, this means that the client no longer needs to get any updates from the registry, but the Seata server should still be running, so EtcdLifeKeeper should not be stopped.

File: discovery/seata-discovery-etcd3/pom.xml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+                <exclusion>
+                    <groupId>com.google.guava</groupId>
+                    <artifactId>guava</artifactId>
+                </exclusion>
+            </exclusions>
+        </dependency>
+        <dependency>
+            <groupId>com.google.guava</groupId>
+            <artifactId>guava</artifactId>
+            <version>27.0.1-jre</version>
```
Comment (github-ygy) [inline discovery/seata-discovery-etcd3/pom.xml]: how about init the vertion in parent pom ？

Comment (xingfudeshi) [reply discovery/seata-discovery-etcd3/pom.xml]: ok.it's done.

**Pull request merged**